### PR TITLE
[feature] Turbo Drive and Sessions in iOS

### PR DESCRIPTION
**Pull request by Pralish**
Addresses: https://github.com/restarone/violet_rails/issues/1414 & https://github.com/restarone/violet_rails/issues/820

### CODE REVIEW

**Comment by donrestarone** [comment]
@Pralish for testing, I cloned the iOS repo: https://github.com/restarone/violet_rails_ios_client and pointed the app root to your review app here: https://github.com/restarone/violet_rails_ios_client/blob/main/Violet%20Rails/Common/Constants/App.swift 

This is definitely an improvement, and I think you can use this technique to debug the server/client faster
Screen Shot 2023-04-23 at 8 54 55 AM

**Comment by donrestarone** [comment]
feedback:

logging in works, but the following links are broken: 

IMG_2760

**Comment by donrestarone** [comment]
# test plan 🧪 

1. HTML Renderer - videos are playable on first visit ⌛ 
2. HTML Renderer - videos playtime is tracked on first visit ⌛ 
3. HTML Renderer - videos playtime is tracked on user-bounce ⌛  
4. Password reset ⌛ 
5. Account creation - with 2FA ⌛ 
6. Account creation - without 2FA ⌛ 
7. Account creation by invite - with 2FA ⌛ 
8. Account creation by invite - without 2FA ⌛ 
9. Sign in - with 2FA ⌛ 
10. Sign in - without 2FA ⌛  
11. Sign out ⌛ 
12. API Form redirect flash message ⌛ 
13. Error flash messages ⌛ 
14. Success flash messages ⌛

**Comment by donrestarone** [comment]
known issues: flash messages are not working on iOS, and 

If we want to enable turbo for the whole app, we need a feature that was introduced in turbo 7.2.0, which is not compatible with node 10

https://github.com/hotwired/turbo/pull/445

Or when we upgrade to rails 7

File: app/views/users/shared/_links.html.erb [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <%= link_to "Log in", new_session_path(resource_name) %><br />
 <% end %>
 
-<%- if devise_mapping.registerable? && controller_name != 'registrations' %>
+<%- if devise_mapping.registerable? && controller_name != 'registrations' && Subdomain.current.allow_user_self_signup %>
```
Comment (donrestarone) [inline app/views/users/shared/_links.html.erb:5]: nice touch

**Comment by donrestarone** [comment]
@Pralish when 2FA is turned off, sign-in does not work
Screen Shot 2023-04-27 at 9 45 02 AM

https://user-images.githubusercontent.com/35935196/234881455-3ea503b0-6825-480e-b4af-523528493568.MP4

This issue will need to be fixed

**Comment by donrestarone** [comment]
@Pralish on first render cookie consent form is broken, this will need to be fixed: 
Screen Shot 2023-04-27 at 9 47 42 AM

**Comment by Pralish** [comment]
Had to rename views with only `.haml` to `.html.haml` for redirect with turbo to work properly. 

Ref: 
https://github.com/hotwired/turbo/issues/138#issuecomment-1518081988

**Comment by Pralish** [comment]
@donrestarone , the issue has been fixed

**Comment by donrestarone** [comment]
⚠️ temporary [deployment](https://github.com/restarone/violet_rails/actions/runs/4670338204/jobs/8618764771) to [nikean.org](https://github.com/restarone/violet_rails/settings/environments/447769648/edit)

**Pull request merged**